vllm/multimodal/base.py

```diff
@@ -188,3 +193,63 @@ def map_input(self, model_config: ModelConfig,
                            f"model class {model_cls.__name__}.")
 
         return mapper(InputContext(model_config), data)
+
+    def register_max_multimodal_tokens(
+        self,
+        max_mm_tokens: MultiModalTokensCalc,
+    ):
+        """
+        Register the maximum number of multi-modal tokens input to the
+        language model for a model class.
+
+        See also:
+            :ref:`adding_a_new_multimodal_model`
+        """
+
+        def wrapper(model_cls: N) -> N:
+            if model_cls in self._input_mappers:
```

Comment:
check in `_max_mm_tokens`?

Reply:
We want `0` to be returned if no input for that modality type is registered, so it is intended to check against `input_mappers` here.

Reply:
We only attempt to calculate `max_mm_tokens` when the input mapper is given.

Reply:
hmm? But should we throw a warning here?
If the input mapper is registered for the model, isn't that one probably also needs to supply with a max token calculation for the model as well through this method?

Reply:
Yeah, that's what the subsequent error is for.

Reply:
Oops, just realized what you mean. I'll fix this.